### PR TITLE
[secure-storage] GitHub Wrapper

### PR DESCRIPTION
**Pull request by davidiw**
This introduces GitHub as another solution to secure-storage and
identifies several other interesting bits along the way:

* GitHub inserts newlines every 60 characters in their base64 encodings -- I don't understand but it means we need to remove them before decoding.
* In order to recursively delete everything, we add a delete_directory and also annotate directories in the "get_directory" query with an extra "/" -- which is exactly how vault does it
* Added a proper liveness check via branch checking
* The actual GitHubStorage is actually rather trivial -- but all tests pass
* Removed a duplicate test on key generation, rotation, exporting

This builds on previous PR: https://github.com/libra/libra/pull/3812

### CODE REVIEW

File: secure/storage/src/github.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{CryptoKVStorage, Error, GetResponse, KVStorage, Value};
+use libra_github_client::Client;
+use libra_secure_time::{RealTimeService, TimeService};
+
+/// GitHubStorage leverages a GitHub repository to provide a file system approach to key / value
+/// storage.  This is not intended for storign private data but for organizing public data.
```
Comment (JoshLind) [inline secure/storage/src/github.rs:9]: nit: storign -> storing

File: secure/storage/github/src/lib.rs [hunk 2]
```diff
@@ -232,8 +279,8 @@ mod tests {
         let value = std::str::from_utf8(&b64_value).unwrap();
         assert_eq!(value, value2);
 
-        // Verify delete
-        github.delete(path).unwrap();
+        // Verify delete_file
```
Comment (zekun000) [inline secure/storage/github/src/lib.rs:282]: does the OWNER, REPOSITORY actually work?

File: secure/storage/src/tests/suite.rs [hunk 3]
```diff
@@ -185,43 +184,6 @@ fn test_create_and_get_non_existent_version(storage: &mut dyn Storage) {
...
-/// are correctly rotated in storage.
-fn test_create_rotate_and_check_key_pair(storage: &mut dyn Storage) {
-    // Create new key pair, fetch both public and private keys and verify relationship
-    let public_key = storage
-        .create_key(CRYPTO_NAME, &Policy::public())
-        .expect("Failed to create a test Ed25519 key pair!");
-    let private_key = storage
-        .export_private_key(CRYPTO_NAME)
-        .expect("Failed to get the private key for a key pair that should exist!");
-    assert_eq!(private_key.public_key(), public_key);
```
Comment (JoshLind) [inline secure/storage/src/tests/suite.rs:257]: nit: if we're going to remove this test, can we add these relationship checks to the test below? i.e., ensure that the public and private keys for a specific key have a relationship? "assert_eq!(private_key.public_key(), public_key);"

File: secure/storage/github/src/lib.rs [hunk 4]
```diff
@@ -79,13 +85,47 @@ impl Client {
...
+                continue;
+            }
+            self.delete_file(&file)?;
+        }
+        Ok(())
+    }
+
+    /// Retrieve a list of branches, this is effectively a status check on the repository
+    pub fn get_branches(&self) -> Result<Vec<String>, Error> {
+        let url = format!("{}/repos/{}/{}/branches", URL, self.owner, self.repository);
```
Comment (zekun000) [inline secure/storage/github/src/lib.rs:103]: this "URL/repose/owner/repository" pattern probably should be abstracted

File: secure/storage/github/src/lib.rs (hunk 2 above)
Comment (davidiw) [reply secure/storage/github/src/lib.rs:282]: Hahaha, you can see the magic: https://github.com/davidiw/secure-store-test

File: secure/storage/github/src/lib.rs (hunk 4 above)
Comment (davidiw) [reply secure/storage/github/src/lib.rs:103]: so then we would potentially call format twice :O

I already have one for general purpose accesses.

**Pull request closed without merging**